determined-ai/determined
# feat: page listing Webhooks [WEB-211]

## Description

At `/det/webhooks`, have a table listing all Webhooks. I modeled the page on ModelRegistry. There isn't pagination, but we could have client-side sort of columns.

We've decided it's "Webhooks" in code and UI text.

For our limited options (Default / Slack) (OnError / OnComplete) I am offering only one selection of these.

## Test Plan

In the DB: 

```sql
insert into webhooks (url, webhook_type) VALUES ('http://example.org', 'DEFAULT');
insert into webhooks (url, webhook_type) VALUES ('http://determined.ai', 'SLACK');

;;  get the id of the webhook created

insert into webhook_triggers (trigger_type, condition, webhook_id) VALUES ('EXPERIMENT_STATE_CHANGE', '{"state":"ERROR"}'::jsonb, 3);
```

- Log in as non-admin user. `/det/webhooks` should not show any webhooks.
- Log in as admin. `/det/webhooks` should display webhooks listed
- Test a webhook (I believe that backend throws an error on this)
- Delete a webhook
- Add a webhook using the create webhook button and modal menu

## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
